### ISSUE TITLE
Connecting diskover-web to AWS ElasticSearch

### ISSUE THREAD

**Issue by DustinHolden** [shirosaidev/diskover-web#4]
When trying to connect diskover-web to an AWS elasticsaerch cluster I am getting the error messages below.

In 'Constants.php' I've set the `ES_HOST` to my VPC endpoint, and can successfully get a curl request back from the endpoint from the instance (to rule out security group issues). `ES_PORT` has been set to 443.

I also have `AWS = TRUE`

Am I missing something simple?

```
PHP Fatal error:  Uncaught Elasticsearch\Common\Exceptions\NoNodesAvailableException: No alive nodes found in your cluster in /home/ubuntu/diskover-web/vendor/elasticsearch/elasticsearch/src/Elasticsearch/ConnectionPool/StaticNoPingConnectionPool.php:51
Stack trace:
#0 /home/ubuntu/diskover-web/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php(72): Elasticsearch\ConnectionPool\StaticNoPingConnectionPool->nextConnection()
#1 /home/ubuntu/diskover-web/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Transport.php(90): Elasticsearch\Transport->getConnection()
#2 /home/ubuntu/diskover-web/vendor/elasticsearch/elasticsearch/src/Elasticsearch/Connections/Connection.php(240): Elasticsearch\Transport->performRequest('HEAD', '/', Array, NULL, Array)
#3 /home/ubuntu/diskover-web/vendor/react/promise/src/FulfilledPromise.php(25): Elasticsearch\Connections\Connection->Elasticsearch\Connections\{closure}(Array)
#4 /home/ubuntu/diskover-web/vendor/guzzlehttp/ringphp/src/Future/CompletedFutureValue.php(55): React\Promise in /home/ubuntu/diskover-web/vendor/elasticsearch/elasticsearch/src/Elasticsearch/ConnectionPool/StaticNoPingConnectionPool.php on line 51
```

The diskover bot/crawler can communicate with ElasticSearch, it is only the web portion having trouble.

**Comment by DustinHolden** [comment]
[https://github.com/jeskew/amazon-es-php/pull/10](https://github.com/jeskew/amazon-es-php/pull/10)

I'm launching an AWS ES cluster that is on 5.3 instead of 5.5 to see if the above issue is the cause.

**Comment by DustinHolden** [comment]
Just an update on 5.3 vs 5.5 on AWS ES. Still received the same error.

**Comment by shirosaidev** [comment]
which version of php and php elasticsearch client are you using? https://github.com/elastic/elasticsearch-php

**Comment by hyattdrew11** [comment]
Getting the exact same error here.

**Comment by shirosaidev** [comment]
I just started testing this bug and it looks like it's related to setting ES_PORT to 443 in Constants.php, setting to 80 fixes this on my test instance of ES 5.5 on AWS.
For the ES_HOST use the endpoint name without http or https.

**Comment by shirosaidev** [comment]
just patched this in v1.5.0-rc4 of diskover-web, let me know if this fixes it for you. Take a look at Constants.php.sample for a new line you will need to add to Constants.php AWS_HTTPS and set to TRUE if you are using https port 443.

**Comment by shirosaidev** [comment]
@DustinHolden and @hyattdrew11 , is this working for you now with aws?
